woocommerce/woocommerce-gutenberg-products-block
# Refactor payment methods components

This PR includes several refactors I did as part of fixing #3166 (I will open follow-up PRs with actual fixes tomorrow). I wanted to extract refactors to a separate PR so they are easier to review and don't get blocked by other changes.

Basically, the rationale of the changes in this PR is:
* Make components smaller so they are easier to work with.
* Remove some `useRef()` usages that didn't seem to be necessary anymore.
* Simplify some of the logic along the way.

### How to test the changes in this Pull Request:

This PR doesn't include any behavior change, so everything should keep working as before.
1. Go to the Checkout block and interact with the payment method tabs. Verify you can toggle them and tab contents always appear.
2. Finish a purchase with one of the tabs and verify the order was done with the correct payment method.

It would be good to test that there are no performance regressions with `trunk`:
* With the devtools Network tab, verify there are no more calls to Stripe API than there were in `trunk`.
* Feel free to add some `console.log()` in some components to verify there are not excessive re-renders compared to `trunk`.
